lib/ipmi_delloem.c

```diff
@@ -338,6 +365,689 @@ usage(void)
+	} else {
+		/* for IPMI 2.0 we get the remaining info from the "get system info" IPMI command rather
+		 * than from the 'get fru data' command */
+
+
+		ipmi_lcd_get_platform_model_name(intf, hostname, IPMI_DELL_LCD_STRING_LENGTH_MAX,
+							IPMI_DELL_SYSINFO_HOST_NAME);
+		printf("Host Name           : %s\n", hostname);
+
+
+		ipmi_lcd_get_platform_model_name(intf, productModel, IPMI_DELL_LCD_STRING_LENGTH_MAX,
+						  IPMI_DELL_PLATFORM_MODEL_NAME_SELECTOR);
+		printf("Product Model       : %s\n", productModel);
+
+		ipmi_sysinfo_get_tag(intf, assettag, IPMI_DELL_SYSINFO_ASSET_TAG,
+										  IPMI_DELL_SYSINFO_ASSET_TAG_LENGTH);
+		printf("Asset Tag           : %s\n", assettag);
+
+		ipmi_sysinfo_get_tag(intf, servicetag, IPMI_DELL_SYSINFO_SERVICE_TAG,
+										  IPMI_DELL_SYSINFO_SERVICE_TAG_LENGTH);
+		printf("Service Tag         : %s\n", servicetag);
+
+		ipmi_lcd_get_platform_model_name(intf, biosversion, IPMI_DELL_LCD_STRING_LENGTH_MAX,
+											 IPMI_DELL_SYSINFO_BIOS_VERSION);
+		printf("BIOS Version        : %s\n", biosversion);
+
+		ipmi_lcd_get_platform_model_name(intf, osName, IPMI_DELL_OS_NAME_LENGTH_MAX,
+											 IPMI_DELL_SYSINFO_OS_NAME);
+		printf("System OS Name      : %s\n", osName);
+
+		ipmi_lcd_get_platform_model_name(intf, osVersion, IPMI_DELL_OS_NAME_LENGTH_MAX,
+											 IPMI_DELL_SYSINFO_OS_VERSION_NUMBER);
+		printf("System OS Version   : %s\n", osVersion);
+
+
+	}
+
+
+	return 0;
+}
+
+/*****************************************************************
+* Function Name: 	ipmi_sysinfo_id
+*
+* Description:	This function prints the 'id' portion of the sysinfo command
+* Input: 		intf 			- pointer to ipmi interface
+*			type			- fru type
+* Output:
+*
+* Return:
+*
+******************************************************************/
+static int
+ipmi_sysinfo_id(struct ipmi_intf * intf, int type)
+{
+	struct ipmi_rs * rsp = NULL;
+	struct ipmi_rq req = {0,0,0,0,0,0};
+	int rc = 0;
+	struct ipm_devid_rsp * dev_id;
+	struct sdr_repo_info_rs * repo_info;
+	uint8_t ipmi_version;
+	uint8_t data[4]={0};
+	struct FW_VER firmware_ver;
+
+	/* 20 80 01 52 51 9f a2 02 00 00 00 00 00 00 00 <- raw data*/
+	memset(&req, 0, sizeof(req));
+
+	if( (iDRAC_FLAG == IDRAC_12G) || (iDRAC_FLAG == IDRAC_13G) || (iDRAC_FLAG == IDRAC_14G) ) {
+		req.msg.netfn = DELL_OEM_NETFN;
+		req.msg.lun = 0;
+		req.msg.cmd = 0xbf;
+		req.msg.data = data;
+		req.msg.data_len = 1;
+		if((iDRAC_FLAG == IDRAC_13G) || (iDRAC_FLAG == IDRAC_14G))
+			data[0] = 0x04;
+		else
+			data[0] = 0x01;
+
+		rsp = intf->sendrecv(intf, &req);
+		if (rsp == NULL) {
+				printf(" Error Getting the Firmware version \n");
+				return -1;
+		}
+		if (rsp->ccode == 0xc1) {
+			memset(&firmware_ver,0,sizeof(struct FW_VER));
+		}
+		else if (rsp->ccode > 0) {
+				printf(" Error getting device ID (%s)\n",
+					val2str(rsp->ccode, completion_code_vals));
+				return -1;
+		}
+		else
+			memcpy(&firmware_ver,(struct FW_VER*)rsp->data,sizeof(struct FW_VER));
```

Comment:
There is no point in casting `rsp->data` to the structure pointer type. `memcpy()` expects `void *`.
Spaces are missing after commas.